### ISSUE TITLE
Default port should be used if `port` is empty/invalid

### ISSUE THREAD

**Issue by cvan** [mattdesl/budo#110]
I want to allow folks to spawn `budo` with a custom default port (instead of `9966`) like this:

```js
PORT=3000 npm start
```

But this won't work:

```json
  "start": "budo --port $PORT"
```

`budo` tries to parse the empty string with `parseInt` and the `NaN` gets passed as an (invalid) port, thereby preventing `budo` from starting the server.

**Comment by yoshuawuyts** [comment]
I'd say this works as intended; if `--port` is passed it's your responsibility to set it. To set a default value regardless try this:
```sh
$ if [ $PORT ]; then budo --port $PORT; else budo; fi
```
I'm not sure what the error looks like you're getting, if it's unclear we could perhaps improve that? Cheers!

**Comment by yoshuawuyts** [comment]
Also to clarify: setting variables before running the script works:
```
 "derp": "echo $foo"
```
```sh
$ foo='bar' npm run foo
bar
```

**Comment by mattdesl** [comment]
Yeah typically CLIs will error out when you pass undefined stuff. I think we could add a clearer error message.

**Comment by cvan** [comment]
I would write the command like this, but I recently discovered Git Bash on Windows doesn't like this format:

```
$ if [ $PORT ]; then budo --port $PORT; else budo; fi
```

And it's also not kind of a pain to type in each project's `npm` `script`s.

## before

### empty
`--port `
<img height="250" alt="screenshot 2015-12-03 02 30 00" src="https://cloud.githubusercontent.com/assets/203725/11559085/3d4a1ec8-996b-11e5-8ad9-420cf7df7f83.png">

### invalid (non-numeric)
`--port swag`
<img height="250" alt="screenshot 2015-12-03 02 37 57" src="https://cloud.githubusercontent.com/assets/203725/11559082/3d2e91ee-996b-11e5-8dec-0743e3a7e121.png">

## after

### valid
`--port 7000`
<img width="1154" alt="screenshot 2015-12-03 03 18 01" src="https://cloud.githubusercontent.com/assets/203725/11561526/e8208966-997c-11e5-997b-4d1c53c31f65.png">

### empty
`--port `
<img width="1154" alt="screenshot 2015-12-03 03 17 02" src="https://cloud.githubusercontent.com/assets/203725/11561527/e820bef4-997c-11e5-996e-def441a1f2c1.png">

### environment variables
`PORT='' … --port $PORT`
`PORT=7000 … --port $PORT`
<img width="1154" alt="screenshot 2015-12-03 03 18 30" src="https://cloud.githubusercontent.com/assets/203725/11561528/e824a212-997c-11e5-8130-b77538825f94.png">

### invalid
`--port swag`
<img width="1154" alt="screenshot 2015-12-03 03 17 29" src="https://cloud.githubusercontent.com/assets/203725/11561529/e8260cd8-997c-11e5-8886-dcfaaeb32622.png">

**Comment by mattdesl** [comment]
Seems like a pretty legitimate use case, I've run into similar issues with Unix stuff not working in Windows.

**Comment by mattdesl** [comment]
Going to reopen this while we figure out another way to allow user-override port/host in a cross-platform manner.
